### PR TITLE
chore(RoleManager): change parameter of create to be one unnamed object

### LINKED ISSUE #5017: Change RoleManager#create parameter
**Is your feature request related to a problem? Please describe.**
The `data` object for the method is the only named parameter among the various `create` methods (for `GuildChannel`s, `Emoji`s, etc).  This inconsistency can lead to confusion for users - "why doesn't my code create a role with the specified name/color/etc?"

**Describe the ideal solution**
Change the parameter to be a single, unnamed object.  This may require putting the `reason` parameter into the object, like it is with `Guild#channels#create`.

### PR DESCRIPTION
**Pull request by ManEatingTapir**
**Please describe the changes this PR makes and why it should be merged:**
Fixes #5017.
This `create` method previously was the only such method with the named parameter `data` in its options. This changes the parameter to be one unnamed object with various properties used in `RoleData` as parameters. These are then passed in the internal call to create the new role with the provided data.
**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/managers/RoleManager.js [hunk 1]
```diff
@@ -87,7 +87,10 @@ class RoleManager extends BaseManager {
    * Creates a new role in the guild with given information.
    * <warn>The position will silently reset to 1 if an invalid one is provided, or none.</warn>
    * @param {Object} [options] Options
-   * @param {RoleData} [options.data] The data to create the role with
+   * @param {string} [options.name] The name of the new role
+   * @param {ColorResolvable} [options.color] The data to create the role with
+   * @param {Permissions} [options.permissions] The permissions for the new role
```
Comment (NotSugden) [inline src/managers/RoleManager.js:92]:
```suggestion
   * @param {PermissionResolvable} [options.permissions] The permissions for the new role
```

File: typings/index.d.ts [hunk 2]
```diff
@@ -2026,7 +2026,13 @@ declare module 'discord.js' {
...
     public guild: Guild;
 
-    public create(options?: { data?: RoleData; reason?: string }): Promise<Role>;
+    public create(options?: {
+      name?: string;
+      color?: ColorResolvable;
+      permissions?: Permissions;
+      position?: number;
+      reason?: string;
+    }): Promise<Role>;
```
Comment (NotSugden) [inline typings/index.d.ts:2035]:
```suggestion
    public create(options?: RoleData & { reason?: string }): Promise<Role>;
```

**Review by NotSugden — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by NotSugden — CHANGES_REQUESTED** [review]
what about `hoist` and `mentionable`? you don't send these here

File: src/managers/RoleManager.js [hunk 3]
```diff
@@ -87,7 +87,10 @@ class RoleManager extends BaseManager {
    * Creates a new role in the guild with given information.
    * <warn>The position will silently reset to 1 if an invalid one is provided, or none.</warn>
    * @param {Object} [options] Options
```
Comment (vladfrangu) [inline src/managers/RoleManager.js:89]: Can't this extend RoleData?

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Not merge ready, see review above this one

File: src/managers/RoleManager.js (hunk 3 above)
Comment (ManEatingTapir) [reply src/managers/RoleManager.js:89]: This is my first PR to the library, as such I am following the pattern of the other managers where they extend BaseManager.

**Comment by ManEatingTapir** [comment]
Added the `hoist` and `mentionable` options to the internal call as requested above, also updated the method documentation to reflect this.

File: src/managers/RoleManager.js (hunk 3 above)
Comment (ManEatingTapir) [reply src/managers/RoleManager.js:89]: Maybe I misunderstood this...Which part could extend RoleData? The RoleManager class, or another part?

Comment (vladfrangu) [reply src/managers/RoleManager.js:89]: The Options object itself, so in the docs, it's just an extension of RoleData

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**